hypothesis-python/tests/cover/test_lookup_py36.py

```diff
@@ -31,3 +32,53 @@ class TreeForwardRefs(typing.NamedTuple):
 @given(st.builds(TreeForwardRefs))
 def test_resolves_forward_references_outside_annotations(t):
     assert isinstance(t, TreeForwardRefs)
+
+
+_ValueType = typing.TypeVar("_ValueType")
+
+
+class Wrapper(typing.Generic[_ValueType]):
+    _inner_value: _ValueType
+
+    def __init__(self, inner_value: _ValueType) -> None:
+        self._inner_value = inner_value
+
+
+@given(st.builds(Wrapper))
+def test_issue_2603_regression(built):
+    """It was impossible to build annotated classes with constructors."""
+    assert isinstance(built, Wrapper)
+
+
+class AnnotatedConstructor(typing.Generic[_ValueType]):
+    value: _ValueType  # the same name we have in `__init__`
+
+    def __init__(self, value: int) -> None:
+        """By this example we show, that ``int`` is more important than ``_ValueType``."""
+        assert isinstance(value, int)
+
+
+@given(st.data())
+def test_constructor_is_more_important(data):
+    """Constructor types should take presence over all other annotations."""
```

Comment:
Oops, I missed this one in review :sweat_smile:

Reply:
👍